pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java

```diff
@@ -805,18 +804,30 @@ public Object visit(ASTCastExpression node, Object data) {
 
     @Override
     public Object visit(ASTPrimaryExpression primaryNode, Object data) {
-        super.visit(primaryNode, data);
+        // visit method arguments in advance
+        for (int i = 0; i < primaryNode.jjtGetNumChildren(); ++i) {
+            ASTArgumentList argList = getArgumentList(primaryNode.jjtGetChild(i)
+                                                              .getFirstChildOfType(ASTArguments.class));
+            if (argList != null) {
+                argList.jjtAccept(this, data);
+            }
+        }
 
         JavaTypeDefinition primaryNodeType = null;
         AbstractJavaTypeNode previousChild = null;
-        AbstractJavaTypeNode nextChild = null;
+        AbstractJavaTypeNode nextChild;
         Class<?> accessingClass = getEnclosingTypeDeclarationClass(primaryNode);
 
         for (int childIndex = 0; childIndex < primaryNode.jjtGetNumChildren(); ++childIndex) {
             AbstractJavaTypeNode currentChild = (AbstractJavaTypeNode) primaryNode.jjtGetChild(childIndex);
             nextChild = childIndex + 1 < primaryNode.jjtGetNumChildren()
                     ? (AbstractJavaTypeNode) primaryNode.jjtGetChild(childIndex + 1) : null;
 
+            // visit non-argument nodes
+            if (currentChild.getFirstChildOfType(ASTArguments.class) == null) {
+                currentChild.jjtAccept(this, data);
```

Comment:
but now we are skipping visitation of all nodes between the current one and the `ASTArgumentList`. (`ASTArguments`, `ASTPrimarySuffix`... maybe something else?)

Why isn't  calling `super.visit(primaryNode, data);` at the beginning of this method as it used to just enough?